lib/util/traverser.js

```diff
@@ -0,0 +1,58 @@
+/**
+ * @fileoverview Wrapper around estraverse
+ * @author Nicholas C. Zakas
+ * @copyright 2016 Nicholas C. Zakas. All rights reserved.
+ * See LICENSE in root directory for full license.
+ */
+"use strict";
+
+//------------------------------------------------------------------------------
+// Requirements
+//------------------------------------------------------------------------------
+
+var estraverse = require("estraverse");
+
+//------------------------------------------------------------------------------
+// Helpers
+//------------------------------------------------------------------------------
+
+var KEY_BLACKLIST = [
+    "parent",
+    "leadingComments",
+    "trailingComments"
+];
+
+/**
+ * Wrapper around an estraverse controller that ensures the correct keys
+ * are visited.
+ * @constructor
+ */
+function Traverser() {
+
+    var controller = Object.create(new estraverse.Controller()),
+        originalTraverse = controller.traverse;
+
+    // intercept call to traverse() and add the fallback key to the visitor
+    controller.traverse = function(node, visitor) {
+        visitor.fallback = Traverser.getKeys;
+        return originalTraverse.call(this, node, visitor);
```

Comment:
I can't comment on expensive, but I am curious why you're using composition instead of prototype inheritance here (which might allow the call stack to shrink by one level if you override traverse rather than wrap it).